File: FitBook.Model/Requests/UserAccounts/UserAccountInsertRequest.cs
namespace FitBook.Model.Requests.UserAccounts;

public class UserAccountInsertRequest
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public string Username { get; set; } = string.Empty;
    public string Password { get; set; } = string.Empty;
    public string Role { get; set; } = string.Empty;
    public string? ProfileImageUrl { get; set; }
    public bool IsActive { get; set; } = true;
}

File: UI/fitbook_desktop/lib/models/requests/user_account_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

import '../common/api_request_body.dart';

part 'user_account_insert_request.g.dart';

/// Mirrors `FitBook.Model.Requests.UserAccounts.UserAccountInsertRequest`.
@JsonSerializable()
class UserAccountInsertRequest implements ApiRequestBody {
  UserAccountInsertRequest({
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.phoneNumber,
    required this.username,
    required this.password,
    required this.role,
    this.profileImageUrl,
    this.isActive = true,
  });

  final String firstName;
  final String lastName;
  final String email;
  final String phoneNumber;
  final String username;
  final String password;
  final String role;
  final String? profileImageUrl;
  final bool isActive;

  factory UserAccountInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$UserAccountInsertRequestFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$UserAccountInsertRequestToJson(this);
}
